### PR TITLE
[PM-5743] Implement eslint rule for usage of window object in background script

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The objective of this PR is to introduce a new eslint rule that requires code changes to remove `window` object references in locations that could potentially not have a valid `window` reference in the global scope. This PR also fixes any existing `window` references where necessary, opting instead to use a `globalThis` or `self` reference where appropriate.

## Code changes

I will just be referencing the code changes that affect the browser extension codebase in a large way. Everything else that is unreferenced represents an update from a `window` reference to a different valid global object reference.

- **apps/browser/.eslintrc.json:** Adding a rule that restricts usage of the global `window` object within the browser extension scope, with specific files excluded.
- **apps/browser/src/popup/services/services.module.ts:** Updating the location of the `BrowserFileDownloadService` to be referenced within a `/popup` route to allow it to use `window` object references. This file is only loaded within the popup.

### CODE REVIEW

**Review by jprusik — DISMISSED** [review]

File: apps/browser/src/autofill/notification/bar.ts
Comment (justindbaur) [inline apps/browser/src/autofill/notification/bar.ts:1]: This is a content script right? Shouldn't it be exempted from the rule?

**Review by justindbaur — COMMENTED** [review]
One question

File: apps/browser/src/autofill/notification/bar.ts
Comment (cagonzalezcs) [reply apps/browser/src/autofill/notification/bar.ts:1]: Yes, that's correct. 

I'd like to keep the updated references to `globalThis`, as that's my preferred approach to referencing the `window` object, [as recommended by OWASP actually.](https://cheatsheetseries.owasp.org/cheatsheets/DOM_Clobbering_Prevention_Cheat_Sheet.html#6-do-not-use-document-and-window-for-global-variables)

Regardless, I've added the scope of this directory and other related autofill content script files to the eslint rule's set of excluded file globs.

**Review by jprusik — DISMISSED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -246,6 +246,24 @@
           }
         ]
       }
+    },
+    {
+      "files": ["apps/browser/src/**/*.ts"],
```
Comment (MGibson1) [inline .eslintrc.json:251]: If the only impacted files live in browser, does it make more sense to put this rule in `apps/browser/.eslintrc.json`?

**Review by MGibson1 — COMMENTED** [review]
One nit about organization of rules

File: .eslintrc.json (hunk 1 above)
Comment (cagonzalezcs) [reply .eslintrc.json]: Good call, I've updated the implementation to reflect your suggestion.

**Review by justindbaur — APPROVED** [review]

**Review by jprusik — APPROVED** [review]

**Pull request merged**